app/models/base.rb

```diff
@@ -307,12 +307,29 @@ def self.parse_datacite_metadata(id: nil, response: nil)
     resource_type_general = attributes.dig("types", "resourceTypeGeneral")
     type = Bolognese::Utils::DC_TO_SO_TRANSLATIONS[resource_type_general.to_s.dasherize] # || attributes.dig("types", "schemaOrg")
 
+    registrant_id = client_id == "crossref.citations" ? get_crossref_member_id(id) : "datacite.#{client_id}"
+
     {
       "@id" => id,
       "@type" => type,
       "datePublished" => get_date(attributes["dates"], "Issued"),
       "proxyIdentifiers" => proxy_identifiers,
-      "registrantId" => "datacite.#{client_id}" }.compact
+      "registrantId" => registrant_id }.compact
+  end
+
+
+  def self.get_crossref_member_id(id, options={})
+    doi = doi_from_url(id)
+    return "crossref.citations" unless doi.present?
+  
+    url = "https://api.crossref.org/works/#{doi}"	
+    # dependency injection
+    response = options[:response] || Maremma.get(url, host: true)	
```

Comment:
Can you explain `options[:response]`?

Reply:
this is for testing. One can inject a mock object of the response rather than relying on the method to make the api call and use vcr. tbf, i ended up not using it but in principles that its intent

Reply:
Thanks. I would remove if you are not using it. As there are other ways to mock this.